chainer/function_hooks/debug_print.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import print_function
+import sys
+
+import chainer
+from chainer import cuda
+from chainer import function
+
+
+class PrintHook(function.FunctionHook):
+    """Function hook that prints debug information.
+
+    Attributes are same as the keyword argument of print function.
```

Comment:
Explain what this hook actually does, and how/when to use. It is also better to compare with just using print function in the forward computation (capability on printing at backward, and printing every function within the `with` statement, right?).